mlir/test/Conversion/ArithToEmitC/arith-to-emitc.mlir

```diff
@@ -48,6 +48,16 @@ func.func @arith_integer_ops(%arg0: i32, %arg1: i32) {
   %3 = arith.divsi %arg0, %arg1 : i32
   // CHECK: emitc.rem %arg0, %arg1 : (i32, i32) -> i32
   %4 = arith.remsi %arg0, %arg1 : i32
+  // CHECK: emitc.bitwise_and %arg0, %arg1 : (i32, i32) -> i32
```

Comment:
I think we should always cast to unsigned types for bitwise operations (except `shrsi`, which is explicitly signed) as
the C11 standard, section 6.5, paragraph 4 states:

```
Some operators (the unary operator ~ , and the binary operators <<, >>, &, ^, and |, collectively described as bitwise operators) shall have operands that have integral type. These operators return values that depend on the internal representations of integers, and thus have implementation-defined and undefined aspects for signed types.
```

Reply:
In this case I'd be inclined to wait for us to bump LLVM before carrying on working on this, as the conversion to unsigned seems to be needed in numerous operations now.